From db1a676814fe9c6ee0273c4b9426982654d80b4f Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 12 Jul 2025 01:10:14 +0800
Subject: [PATCH] Use _DARWIN_C_SOURCE on macOS, not _GNU_SOURCE

---
 src/compat.c             | 4 ++++
 tools/config/configure.R | 8 ++++++++
 2 files changed, 12 insertions(+)

diff --git a/src/compat.c b/src/compat.c
index 7c99a3f..bcf3283 100644
--- src/compat.c
+++ src/compat.c
@@ -1,5 +1,9 @@
 #define _POSIX_C_SOURCE 200809L
+#ifdef __APPLE__
+#define _DARWIN_C_SOURCE
+#else
 #define _GNU_SOURCE
+#endif
 #define _CRT_RAND_S
 #define _CRT_NONSTDC_NO_DEPRECATE
 
diff --git a/tools/config/configure.R b/tools/config/configure.R
index 7083852..1a17f44 100644
--- tools/config/configure.R
+++ tools/config/configure.R
@@ -31,7 +31,11 @@ if (.Platform$OS.type != "windows") {
   tmpl <- "
 #pragma GCC diagnostic error \"-Wimplicit-function-declaration\"
 #define _POSIX_C_SOURCE 200809L
+#ifdef __APPLE__
+#define _DARWIN_C_SOURCE
+#else
 #define _GNU_SOURCE
+#endif
 #include <stdlib.h>
 int f() {
   return arc4random();
@@ -45,7 +49,11 @@ int f() {
   tmpl <- "
 #pragma GCC diagnostic error \"-Wimplicit-function-declaration\"
 #define _POSIX_C_SOURCE 200809L
+#ifdef __APPLE__
+#define _DARWIN_C_SOURCE
+#else
 #define _GNU_SOURCE
+#endif
 #include <unistd.h>
 int f() {
   unsigned int u;
