--- NAMESPACE	2023-09-29 00:11:50.000000000 +0800
+++ NAMESPACE	2025-06-19 11:04:42.000000000 +0800
@@ -43,8 +43,8 @@
 importFrom(ggplot2,theme_minimal)
 importFrom(glue,glue)
 importFrom(mvtnorm,pmvnorm)
-importFrom(optimizeR,apply_optimizer)
-importFrom(optimizeR,optimizer_nlm)
+# importFrom(optimizeR,apply_optimizer)
+# importFrom(optimizeR,optimizer_nlm)
 importFrom(parallel,makeCluster)
 importFrom(parallel,stopCluster)
 importFrom(reshape2,melt)
