--- configure	2025-05-20 14:52:55.000000000 +0800
+++ configure	2025-05-23 08:07:00.000000000 +0800
@@ -42,10 +42,10 @@
   PKG_LIBS="-L$LIB_DIR -lmbedtls -lmbedx509 -lmbedcrypto $PKG_LIBS"
   echo "Found INCLUDE_DIR $INCLUDE_DIR"
   echo "Found LIB_DIR $LIB_DIR"
-elif [ -d "/usr/local/include/mbedtls" ]
+elif [ -d "@PREFIX@/include/mbedtls" ]
 then
-  PKG_CFLAGS="-I/usr/local/include $PKG_CFLAGS"
-  PKG_LIBS="-L/usr/local/lib -lmbedtls -lmbedx509 -lmbedcrypto $PKG_LIBS"
+  PKG_CFLAGS="-I@PREFIX@/include $PKG_CFLAGS"
+  PKG_LIBS="-L@PREFIX@/lib -lmbedtls -lmbedx509 -lmbedcrypto $PKG_LIBS"
   echo "Found 'libmbedtls' $PKG_CFLAGS"
 elif [ -d "/usr/include/mbedtls" ]
 then
@@ -114,10 +114,10 @@
 then
   NNG_CFLAGS="-I$INCLUDE_DIR"
   NNG_LIBS="-L$LIB_DIR -lnng"
-elif [ -d "/usr/local/include/nng" ]
+elif [ -d "@PREFIX@/include/nng" ]
 then
-  NNG_CFLAGS="-I/usr/local/include"
-  NNG_LIBS="-L/usr/local/lib -lnng"
+  NNG_CFLAGS="-I@PREFIX@/include"
+  NNG_LIBS="-L@PREFIX@/lib -lnng"
 elif [ -d "/usr/include/nng" ]
 then
   NNG_CFLAGS="-I/usr/include"
