From a617eaf5b848b9afaf87baffafe7cc80576bad1a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 9 Feb 2023 01:41:55 +0700
Subject: [PATCH 3/3] libuv: link to legacysupport

---
 src/Makevars | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makevars b/src/Makevars
index dd892c7d..6eb0a5cc 100644
--- a/src/Makevars
+++ b/src/Makevars
@@ -45,7 +45,7 @@
 $(LIBUV)/Makefile:
 	touch $(LIBUV)/aclocal.m4 && touch $(LIBUV)/configure && touch $(LIBUV)/Makefile.in
 	(cd $(LIBUV) \
-	&& CC="$(CC)" CPPFLAGS="$(CPPFLAGS)" CFLAGS="$(CFLAGS) $(CPICFLAGS) $(C_VISIBILITY) -std=c99" AR="$(AR)" RANLIB="$(RANLIB)" LDFLAGS="$(LDFLAGS)" LT_MULTI_MODULE=1 ./configure $(R_CONFIGURE_FLAGS) $(CONFIGURE_ARGS) --quiet)
+	&& CC="$(CC)" CPPFLAGS="$(CPPFLAGS)" CFLAGS="$(CFLAGS) @MP_FLAGS@ $(CPICFLAGS) $(C_VISIBILITY) -std=c99" AR="$(AR)" RANLIB="$(RANLIB)" LDFLAGS="$(LDFLAGS) -lMacportsLegacySupport" LT_MULTI_MODULE=1 ./configure $(R_CONFIGURE_FLAGS) $(CONFIGURE_ARGS) --quiet)
 
 $(LIBUV)/.libs/libuv.pa: $(LIBUV)/Makefile
 	$(MAKE) --directory=$(LIBUV) \
